modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/metadata/NodesMapping.java

```diff
@@ -52,6 +52,9 @@
     /** */
     public static final byte DEDUPLICATED           = 1 << 4;
 
+    /** */
+    public static final byte CLIENT                 = 1 << 5;
+
     /** */
     private final List<UUID> nodes;
 
```

Comment:
It looks so because partitions assignments implemented the same way. There is a trade off between calculation complexity and memory allocations. I'll think how to improve it. Note that it's still one of first naive implementations I came up with.

Reply:
It would be cool if we have a jira ticket for refactoring and simplification of this class

Reply:
OK. I'll create it.

Reply:
https://issues.apache.org/jira/browse/IGNITE-12585

Reply:
https://issues.apache.org/jira/browse/IGNITE-12586